### PR TITLE
Big set scenario

### PR DESCRIPTION
**Pull request by aduprat**
Replace: #36

### CODE REVIEW

**Review by chibenwa — APPROVED** [review]

File: src/main/scala-2.11/org/apache/james/gatling/jmap/JmapMailboxes.scala [hunk 1]
```diff
@@ -29,6 +29,7 @@ object JmapMailboxes {
   private val inboxIdPath = s"$$$mailboxListPath[?(@.role == 'inbox')].id"
   private val outboxIdPath = s"$$$mailboxListPath[?(@.role == 'outbox')].id"
   private val sentIdPath = s"$$$mailboxListPath[?(@.role == 'sent')].id"
+  val numberOfSystemMailboxes = 5
```
Comment (rouazana) [inline src/main/scala-2.11/org/apache/james/gatling/jmap/JmapMailboxes.scala:31]: not used?

File: src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/JmapBigSetScenario.scala [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  val scn = scenario("JmapBigSet")
+    .exec(CommonSteps.provisionUsersWithMailboxesAndMessages(users, NumberOfMailboxes, NumberOfMessages))
+    .during(ScenarioDuration) {
+      execWithRetryAuthentication(JmapMailboxes.getMailboxes, JmapMailboxes.getMailboxesChecks(numberOfMailboxesPerUser))
+        .exec(execWithRetryAuthentication(JmapMessages.listMessages, JmapMessages.listMessagesChecks))
+        .pause(1 second , 2 seconds)
+    }
+
+  def numberOfMailboxesPerUser: Int =
+    NumberOfMailboxes + numberOfSystemMailboxes
```
Comment (rouazana) [inline src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/JmapBigSetScenario.scala:41]: oh it is :)

**Review by rouazana — APPROVED** [review]

File: build.sbt [hunk 3]
```diff
@@ -10,9 +10,11 @@ enablePlugins(GatlingPlugin)
...
+resolvers += "Fabricator" at "http://dl.bintray.com/biercoff/Fabricator"
 
 libraryDependencies += "com.typesafe.play" %% "play-ws" % "2.4.3"
 libraryDependencies += "io.gatling" % "gatling-test-framework" % "2.2.2"
 libraryDependencies += "io.gatling.highcharts" % "gatling-charts-highcharts" % "2.2.2"
 
-libraryDependencies += "ch.lightshed" %% "courier" % "0.1.4"
\ No newline at end of file
+libraryDependencies += "ch.lightshed" %% "courier" % "0.1.4"
+libraryDependencies += "com.github.azakordonets" % "fabricator_2.11" % "2.1.5"
```
Comment (mbaechler) [inline build.sbt]: using %% allow to omit _2.11

File: src/main/scala-2.11/org/apache/james/gatling/jmap/JmapMailboxes.scala [hunk 4]
```diff
@@ -5,6 +5,23 @@ import io.gatling.http.Predef._
 import io.gatling.http.check.HttpCheck
 import org.apache.james.gatling.utils.JmapChecks
 import org.apache.james.gatling.utils.RetryAuthentication.execWithRetryAuthentication
+import org.apache.james.gatling.utils.RandomStringGenerator
+import fabricator.Words
+import scala.concurrent.Future
```
Comment (mbaechler) [inline src/main/scala-2.11/org/apache/james/gatling/jmap/JmapMailboxes.scala]: where it is used ?

File: src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/JmapBigSetScenario.scala [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+/*
+ * The aim of the scenario is to provide multiple mailboxes per user and several mails in them.
+ * Then, it's just checking that the created mailboxes exists.
+ */
+class JmapBigSetScenario extends Simulation {
+
+  val users = new UserCreator(BaseJamesWebAdministrationUrl).createUsersWithInboxAndOutbox(UserCount)
+  
+  val scn = scenario("JmapBigSet")
```
Comment (mbaechler) [inline src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/JmapBigSetScenario.scala]: could we have a more descriptive sentence ?

File: src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/common/CommonSteps.scala [hunk 6]
```diff
@@ -36,6 +39,23 @@ object CommonSteps {
       }
       .pause(30 second)
 
+  def provisionUsersWithMailboxesAndMessages(users: Seq[Future[User]], numberOfMailboxes: Int, numberOfMessages: Int) = {
+    scenario("ProvisionUsersWithMailboxesAndMessages")
+      .exec(provisionSystemMailboxes(users))
+      .repeat(numberOfMailboxes) {
+        exec((session: Session) => session.set("createdId", Id.generate.id))
```
Comment (mbaechler) [inline src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/common/CommonSteps.scala]: please extract the 3 next steps to a single method

**Review by mbaechler — APPROVED** [review]

**Pull request merged**